### PR TITLE
sql: add telemetry for ON UPDATE

### PR DESCRIPTION
**Pull request by ZhouXing19**
This commit is to add telemetry for a column
created / altered with `ON UPDATE` syntax.

Release Note: None

### CODE REVIEW

**Review by RichardJCai — APPROVED** [review]
LGTM, thanks for doing this!

File: pkg/sql/create_table.go [hunk 1]
```diff
@@ -2842,6 +2842,9 @@ func incTelemetryForNewColumn(def *tree.ColumnTableDef, desc *descpb.ColumnDescr
 			telemetry.Inc(sqltelemetry.SchemaNewColumnTypeQualificationCounter("unique"))
 		}
 	}
+	if desc.HasOnUpdate() {
```
Comment (rafiss) [inline pkg/sql/create_table.go:2845]: does this code also get used for ALTER TABLE statements?

Comment (ZhouXing19) [reply pkg/sql/create_table.go:2845]: Yep, check [here](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/add_column.go#L75).

**Review by rafiss — APPROVED** [review]
thanks!

**Pull request merged**